spec/legacy/unit/journal_spec.rb

```diff
@@ -37,21 +37,19 @@
     end
   end
 
-  it 'should create should send email notification' do
-    ActionMailer::Base.deliveries.clear
+  it 'create should send email notification' do
     issue = WorkPackage.find(:first)
     if issue.journals.empty?
       issue.add_journal(User.current, 'This journal represents the creationa of journal version 1')
       issue.save
     end
-    user = User.find(:first)
-    assert_equal 0, ActionMailer::Base.deliveries.size
```

Comment:
I've digged through that particular spec a bit (even though my previous knowledge of the journal is limited). What I've found was that on `release/4.3`, `Setting.notified_events.include?('work_package_added')` was set to false, thus the whole notification in `handle_work_package_create` was not evaluated, and no deliveries were made.

In your changes, the `send_notification?` evaluates to true due to
* `(Setting.notified_events.include?('work_package_added') && journal.initial?)` being false (the former unchanged being false as in release/4.3)
* `Setting.notified_events.include?('work_package_updated')` **being true**

I'm not sure if this is a big issue in the updated code, but I guess you'd have a better insight here to tell if that makes a significant difference.